controllers/pkg/giteaclient/giteaclient.go

```diff
@@ -36,10 +37,25 @@ type GiteaClient interface {
 	Get() *gitea.Client
 }
 
-func New(client resource.APIPatchingApplicator) GiteaClient {
-	return &gc{
-		client: client,
+var lock = &sync.Mutex{}
+
+var singleInstance *gc
+
+func New(client resource.APIPatchingApplicator, ctx context.Context) GiteaClient {
```

Comment:
go best practice is to always add ctx as a first arg in a fn.

Reply:
Thanks for pointing out, will update the PR